packages/expo-image/android/src/main/java/expo/modules/image/ExpoImageView.java

```diff
@@ -0,0 +1,146 @@
+import expo.modules.image.events.ImageErrorEvent;
+import expo.modules.image.events.ImageLoadEndEvent;
+import expo.modules.image.events.ImageLoadEvent;
+import expo.modules.image.events.ImageLoadStartEvent;
+import expo.modules.image.events.ImageProgressEvent;
+import expo.modules.image.okhttp.OkHttpClientProgressInterceptor;
+import expo.modules.image.okhttp.OkHttpClientResponseInterceptor;
+import expo.modules.image.okhttp.ResponseListener;
+import okhttp3.MediaType;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+
+@SuppressLint("ViewConstructor")
+public class ExpoImageView extends AppCompatImageView implements ResponseListener, ProgressListener {
+  private static final String SOURCE_URI_KEY = "uri";
+
+  private OkHttpClientProgressInterceptor mProgressInterceptor;
+  private OkHttpClientResponseInterceptor mResponseInterceptor;
+  private RequestManager mRequestManager;
+  private ReadableMap mSourceMap;
+  private GlideUrl mLoadedSource;
+  private MediaType mMediaType;
+  private RCTEventEmitter mEventEmitter;
+
+  public ExpoImageView(ReactContext context, RequestManager requestManager, OkHttpClientProgressInterceptor progressInterceptor, OkHttpClientResponseInterceptor responseInterceptor) {
+    super(context);
+
+    mEventEmitter = context.getJSModule(RCTEventEmitter.class);
+    mRequestManager = requestManager;
+    mProgressInterceptor = progressInterceptor;
+    mResponseInterceptor = responseInterceptor;
+
+    // For now let's set scale type to FIT_XY
+    // to make behavior same on all platforms.
+    setScaleType(ImageView.ScaleType.FIT_XY);
+  }
+
+  /* package */ void setSource(@Nullable ReadableMap sourceMap) {
+    mSourceMap = sourceMap;
+  }
+
+  /* package */ void onAfterUpdateTransaction() {
+    GlideUrl sourceToLoad = createUrlFromSourceMap(mSourceMap);
+
+    if (sourceToLoad == null) {
+      mRequestManager.clear(this);
+      setImageDrawable(null);
+      mLoadedSource = null;
+    } else if (!sourceToLoad.equals(mLoadedSource)) {
+      mLoadedSource = sourceToLoad;
+      mMediaType = null;
+      new ImageLoadStartEvent(getId()).dispatch(mEventEmitter);
+      mResponseInterceptor.registerResponseListener(sourceToLoad.toStringUrl(), this);
+      mProgressInterceptor.registerProgressListener(sourceToLoad.toStringUrl(), this);
+      mRequestManager
+          .load(sourceToLoad)
+          .listener(createRequestListener())
+          .into(this);
+
+    }
+  }
+
+  /* package */ void onDrop() {
+    mRequestManager.clear(this);
+  }
+
+  @Nullable
+  protected GlideUrl createUrlFromSourceMap(@Nullable ReadableMap sourceMap) {
+    if (sourceMap == null || sourceMap.getString(SOURCE_URI_KEY) == null) {
+      return null;
+    }
+
+    return new GlideUrl(sourceMap.getString(SOURCE_URI_KEY));
+  }
+
+  @Override
+  public void onResponse(Response response) {
+    ResponseBody body = response.body();
+    if (body != null) {
+      mMediaType = body.contentType();
+    }
```

Comment:
The `mediaType` in the tests always returns `null` on Android. Could it be this code is only called when downloading a response from the network ?

![image](https://user-images.githubusercontent.com/6184593/76845954-050aa200-6840-11ea-9aeb-999f2b238db8.png)

Reply:
That is correct. I haven't found a way to mitigate this, unfortunately. 😕

Reply:
Mitigated by a separate call to Glide decoding `InputStream` as `BitmapFactory.Options` (with `inJustDecodeBounds` set to `true` so we don't really decode the stream to bitmap).